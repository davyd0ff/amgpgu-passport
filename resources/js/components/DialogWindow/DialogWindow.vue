<style scoped>
.modal.open {
  display: flex !important;
  flex-direction: column;
  overflow: hidden;
}

.modal-header {
  position: relative;
}

.modal-header .input-field {
  margin: 10px;
  margin-bottom: 0;
}

.modal-content {
  overflow: auto;
  position: relative;
}

.modal-footer {
  position: relative;
}
</style>

<template>
  <div
    ref="modal"
    id="dialog-window"
    class="modal"
    v-bind:class="{ open: isOpened }"
  >
    <div class="modal-header">
      <slot name="header" />
    </div>
    <div class="modal-content">
      <slot name="content" />
    </div>
    <div class="modal-footer">
      <a
        href="#!"
        class="modal-close waves-effect waves-green btn"
        v-on:click.prevent
      >
        {{ 'DIALOG_TREE_SELECT' | localize }}
      </a>
      <!-- <slot name="footer" /> -->
    </div>
  </div>
</template>

<script>
// import { defineComponent } from '@vue/composition-api';
import M from 'materialize-css';

export default {
  name: 'DialogWindow',
  props: {
    isOpened: { type: Boolean, default: false },
  },
  // watch: {
  //   isOpened: function (value) {
  //     if (value && this.$modal) {
  //       this.$modal.open();
  //     }
  //   },
  // },
  mounted() {
    this.$modal = M.Modal.init(this.$refs.modal, {
      onCloseStart: this.onClose,
    });
    this.openDialog();
  },
  beforeUpdate() {
    this.openDialog();
  },
  beforeDestroy() {
    if (this.$modal) {
      this.$modal.destroy();
    }
  },
  methods: {
    openDialog: function () {
      if (this.isOpened && this.$modal) {
        this.$modal.open();
      }
    },
    onClose: function () {
      this.$emit('on-close');
    },
  },
};
</script>
