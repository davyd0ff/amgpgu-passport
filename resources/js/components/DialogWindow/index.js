import DialogWindow from './DialogWindow.vue';

export default DialogWindow;
