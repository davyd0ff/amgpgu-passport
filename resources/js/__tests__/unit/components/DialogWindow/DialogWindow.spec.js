import { shallowMount, createLocalVue } from '@vue/test-utils';
import DialogWindow from '@/components/DialogWindow/DialogWindow.vue';

describe('DialogWindow', () => {
  const makeWrapper = (options) => {
    const localize = (str) => str;
    const localVue = createLocalVue();
    localVue.filter('localize', localize);

    return shallowMount(DialogWindow, { ...options, localVue });
  };

  it('render with slots', () => {
    const wrapper = makeWrapper({
      slots: {
        header: '<div id="test-slot-header" />',
        content: '<div id="test-slot-content" />',
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('render when prop IsOpened is truthy', () => {
    const wrapper = makeWrapper({
      propsData: {
        isOpened: true,
      },
    });

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('#dialog-window').classes()).toContain('open');
  });

  it('render when prop IsOpened is falsy', () => {
    const wrapper = makeWrapper({
      propsData: {
        isOpened: false,
      },
    });

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('#dialog-window').classes()).not.toContain('open');
  });
});
