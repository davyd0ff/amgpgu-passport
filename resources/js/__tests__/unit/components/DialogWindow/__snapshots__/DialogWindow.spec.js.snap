// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogWindow render when prop IsOpened is falsy 1`] = `
<div id="dialog-window" class="modal" tabindex="0">
  <div class="modal-header"></div>
  <div class="modal-content"></div>
  <div class="modal-footer"><a href="#!" class="modal-close waves-effect waves-green btn">
      DIALOG_TREE_SELECT
    </a></div>
</div>
`;

exports[`DialogWindow render when prop IsOpened is truthy 1`] = `
<div id="dialog-window" class="modal open" tabindex="0" style="z-index: 1003; display: block; opacity: 0; top: 4%; transform: scaleX(0.8) scaleY(0.8);">
  <div class="modal-header"></div>
  <div class="modal-content"></div>
  <div class="modal-footer"><a href="#!" class="modal-close waves-effect waves-green btn">
      DIALOG_TREE_SELECT
    </a></div>
</div>
`;

exports[`DialogWindow render with slots 1`] = `
<div id="dialog-window" class="modal" tabindex="0">
  <div class="modal-header">
    <div id="test-slot-header"></div>
  </div>
  <div class="modal-content">
    <div id="test-slot-content"></div>
  </div>
  <div class="modal-footer"><a href="#!" class="modal-close waves-effect waves-green btn">
      DIALOG_TREE_SELECT
    </a></div>
</div>
`;
